app/src/main/java/com/infomaniak/swisstransfer/ui/utils/PermissionsUtils.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2025 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.utils
+
+import android.os.Build
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.isGranted
+
+@OptIn(ExperimentalPermissionsApi::class)
+fun PermissionState?.checkPermission(action: () -> Unit) {
+    if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.P && this?.status?.isGranted != true) {
```

Comment:
@KevinBoulongne I forgot to mention that this condition is dangerous, because this function might be used in other parts of the app, and then, it'd break for API levels greater than "P" (BTW, can we use literal integers for that, please 🙏 )